src/validators.ts

```diff
@@ -114,3 +114,21 @@ export function optional<T>(
     return validator(val);
   };
 }
+
+export function optionalArray<T>(validator: Validator<T>): Validator<T | T[]>;
```

Comment:
Elsewhere `optional` means `T | undefined` so at a glance I would expect `optionalArray(...)` to be the same as `optional(array(...))`. So I'd prefer to use a different term here. How about `maybeArray`?

Reply:
That's reasonable, I  update name accordingly, Thanks